tests/integration/ceph_smoke_test.go

```diff
@@ -341,3 +343,110 @@ func (s *SmokeSuite) getNonCanaryMonDeployments() ([]appsv1.Deployment, error) {
 	}
 	return nonCanaryMonDeployments, nil
 }
+
+// Smoke Test for Webhooks
+func (s *SmokeSuite) TestWebhooks() {
+	logger.Info("Create webhook Smoke Test")
+	ctx := context.TODO()
+
+	cephCRList, err := s.k8sh.RookClientset.CephV1().CephClusters(s.settings.Namespace).List(ctx, metav1.ListOptions{})
+	require.NoError(s.T(), err)
+	cephCR := cephCRList.Items[0]
+	cephCR.Spec.DataDirHostPath = "/var"
+	t := s.T()
+
+	t.Run("Fail updating cephCluster, hostPath cannot be changes", func(t *testing.T) {
+		_, err = s.k8sh.RookClientset.CephV1().CephClusters(s.settings.Namespace).Update(ctx, &cephCR, metav1.UpdateOptions{})
+		require.Error(s.T(), err)
+	})
+
+	pool := &cephv1.CephBlockPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "webhook-pool",
+			Namespace: s.settings.Namespace,
+			UID:       types.UID("c47cac40-9bee-4d52-823b-ccd803ba5bfe"),
+		},
+		Spec: cephv1.NamedBlockPoolSpec{
+			PoolSpec: cephv1.PoolSpec{
+				Replicated: cephv1.ReplicatedSpec{
+					Size: uint(1),
+				},
+				ErasureCoded: cephv1.ErasureCodedSpec{
+					CodingChunks: 1,
+					DataChunks:   1,
+				},
+			},
+		},
+	}
+
+	t.Run("Fail pool creation, both erasurecoded and replicated fields cannot be set at the same time", func(t *testing.T) {
+		_, err = s.k8sh.RookClientset.CephV1().CephBlockPools(s.settings.Namespace).Create(ctx, pool, metav1.CreateOptions{})
+		require.Error(s.T(), err)
+	})
```

Comment:
```suggestion
	t.Run("Fail pool creation, both erasurecoded and replicated fields cannot be set at the same time", func(t *testing.T) {
	pool := &cephv1.CephBlockPool{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "webhook-pool",
			Namespace: s.settings.Namespace,
			UID:       types.UID("c47cac40-9bee-4d52-823b-ccd803ba5bfe"),
		},
		Spec: cephv1.NamedBlockPoolSpec{
			PoolSpec: cephv1.PoolSpec{
				Replicated: cephv1.ReplicatedSpec{
					Size: uint(1),
				},
				ErasureCoded: cephv1.ErasureCodedSpec{
					CodingChunks: 1,
					DataChunks:   1,
				},
			},
		},
	}
	_, err = s.k8sh.RookClientset.CephV1().CephBlockPools(s.settings.Namespace).Create(ctx, pool, metav1.CreateOptions{})
	require.Error(s.T(), err)
```

My suggestion was something like the above for the pool example.

Reply:
ah, got it now. Thanks